File: c-sharp/Models/VigenereText.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Ciphers.Models
{
    public class VigenereText : Text
    {
        public string KeyWord { get; set; }

        public VigenereText(string plainTxt, string keyWord) : base(plainTxt)
        {
            KeyWord = keyWord;
        }
    }
}

File: java/src/main/java/com/metatarsal/ciphers/models/VigenereText.java
package com.metatarsal.ciphers.models;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode()
public class VigenereText extends Text {

    private String keyword;

    public VigenereText(String plainText, String keyword) {
        super();
        this.plainText = plainText;
        this.keyword = keyword;
    }

    public String getKeyword() {
        return keyword;
    }
}
